From 60f23d71faf57b7a096af4b55280cdca2dfc7b53 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Wed, 5 Jul 2023 17:29:48 +0200
Subject: [PATCH 0398/1141] memory: stm32_omi: rework stm32_omi_dlyb_stop API

Reset DLYB CR register.

Change-Id: I2d101ffdc84f6448a35547b6009f7766bfad2732
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/316856
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
Domain-Review: Patrice CHOTARD <patrice.chotard@foss.st.com>
---
 drivers/memory/stm32-omi.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/memory/stm32-omi.c b/drivers/memory/stm32-omi.c
index c7d352d70f85..199e6d872ac5 100644
--- a/drivers/memory/stm32-omi.c
+++ b/drivers/memory/stm32-omi.c
@@ -470,10 +470,8 @@ static u32 stm32_omi_find_byp_cmd(u16 period_ps)
 
 void stm32_omi_dlyb_stop(struct stm32_omi *omi)
 {
-	/* disable delay block and bypass mode */
-	regmap_update_bits(omi->regmap,
-				 omi->dlyb_base + SYSCFG_DLYBOS_CR,
-				 DLYBOS_CR_EN | DLYBOS_BYP_EN, 0);
+	/* disable delay block */
+	regmap_write(omi->regmap, omi->dlyb_base + SYSCFG_DLYBOS_CR, 0x0);
 }
 EXPORT_SYMBOL(stm32_omi_dlyb_stop);
 
-- 
2.39.2

